cmake/SerialboxConfig.cmake.in

```diff
@@ -40,12 +40,78 @@ set(SERIALBOX_BOOST_LIBRARY_DIRS "@Boost_LIBRARY_DIRS@")
 set(SERIALBOX_BOOST_INCLUDE_DIRS "@Boost_INCLUDE_DIRS@")
 set(SERIALBOX_REQUIRED_BOOST_COMPONENTS "@REQUIRED_BOOST_COMPONENTS@")
 
+#===---------------------------------------------------------------------------------------------===
+#   Helper functions used in this file
+#====--------------------------------------------------------------------------------------------===
+
+function( serialbox_print_targets )
+  #
+  # serialbox_print_targets( targets_list )
+  #
+  #    Print exported and unavailable targets
+  #
+  message( STATUS "Serialbox exported targets:")
+  foreach( target ${ARGV} )
+    if( TARGET ${target} )
+      message( STATUS "  ${target}" )
+    else()
+      list( APPEND _unavailable_targets ${target} )
+    endif()
+  endforeach()
+  if( _unavailable_targets )
+    message( STATUS "Serialbox unavailable targets:")
+    foreach( target ${_unavailable_targets} )
+      message( STATUS "  ${target}" )
+    endforeach()
+  endif()
+endfunction()
+
+function( serialbox_alias_targets )
+  #
+  # serialbox_alias_targets( alias target_shared target_static align )
+  #
+  #    Creates ALIAS target for given targets. Prefer target_shared if availabe
+  # 
+  set( _alias ${ARGV0} )
+  set( _target_shared ${ARGV1} )
+  set( _target_static ${ARGV2} )
+  set( _align ${ARGV3} )
+  string( REPLACE "Serialbox::" "" _target_shared_build "${_target_shared}" )
+  string( REPLACE "Serialbox::" "" _target_static_build "${_target_static}" )
+  if( TARGET ${_target_shared_build} )
```

Comment:
Why is this necessary? You also added the `serialbox_install_targets()` which should have been called in the case of `if( TARGET ${_target_shared_build} )`, so no need to check here if the alias exists?

Reply:
I wanted to create the aliases
- Serialbox::Serialbox_C   --> SerialboxCShared or SerialboxCStatic
- Serialbox::Serialbox_Fortran --> SerialboxFortranShared or SerialboxFortranStatic

which prefer the shared build if available, and fall-back to static. It's always better to use shared because Serialbox2 dependencies can stay private. Using the aliases, client code needs not to analyze if shared is available. And if static is what you wanted you can always link to Serialbox::SerialboxCStatic as it is always available

For `Serialbox::Serialbox_C`:

If 1) `SerialboxCShared` or 2) `Serialbox::SerialboxCShared` are a target then an alias is created if not available yet.

In case of 1) we are in a super-build configuration and the target `Serialbox::SerialboxCShared` target exists. But CMake disallows to alias an alias. Therefore we need to alias the original target
```
Serialbox::SerialboxC --> SerialboxCShared
```
In case of 2) Serialbox is from a separate install-tree.  `SerialboxCShared` does not exist as target, but `Serialbox::SerialboxCShared` exists as **IMPORTED** target. It ***is*** allowed to alias an IMPORTED target, provided that we set the PROPERTY ```IMPORTED_GLOBAL```